### PR TITLE
UIDATIMP-722: Cover <RepeatableActionsField> component with unit tests

### PR DESCRIPTION
**Pull request by Javokhir12**
## Purpose

Cover RepeatActionsField component with tests using RTL+Jest.
## Approach

Covered RepeatActionsField component with unit tests using Jest + RTL

## Refs

https://issues.folio.org/browse/UIDATIMP-722
## Screenshots

Screenshot (18)

### CODE REVIEW

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  actionToClearFields: REPEATABLE_ACTIONS.DELETE_EXISTING,
+  subfieldsToClearPath: '',
+  repeatableFieldIndex: 15,
+  hasRepeatableFields: true,
+  repeatableFieldAction: REPEATABLE_ACTIONS.EXTEND_EXISTING,
+  onRepeatableActionChange: noop,
+};
+
+describe('RepeatableActionsField component', () => {
+  test('should be rendered with child component', () => {
```
Comment (OleksandrHladchenko1) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:80]: Here and in the next testcases, please, replace **test** with **it**, for consistency

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    const { getByText } = renderRepeatableActionsField({ ...RepeatableActionsFieldProps });
+
+    screen.debug();
+    expect(getByText('Repeatable field component'))
+      .toBeDefined();
+  });
+  describe('when click on Select action, if selected action and clear action are equal', () => {
```
Comment (OleksandrHladchenko1) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:87]: Please, separate this long describe into two short

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      subfieldsToClearPath={subfieldsToClearPath}
+    >
+      {props => childComponent(props)}
+    </RepeatableActionsField>
+  );
+
+  return renderWithIntl(renderWithReduxForm(component), translationsProperties);
+};
+
+const RepeatableActionsFieldProps = {
```
Comment (OleksandrHladchenko1) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:65]: Please, declare this variable with a small letter

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import { noop } from 'lodash';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+
+import '../../../test/jest/__mock__';
+
+import { fireEvent } from '@testing-library/react';
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:9]: Move this to `react` imports, please
See code style for imports sequence here - https://wiki.folio.org/pages/resumedraft.action?draftId=14458784&draftShareId=483fa2fb-615e-4002-a916-c976bd5fb8c6&

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      subfieldsToClearPath={subfieldsToClearPath}
+    >
+      {props => childComponent(props)}
+    </RepeatableActionsField>
+  );
+
+  return renderWithIntl(renderWithReduxForm(component), translationsProperties);
+};
+
+const repeatableActionsFieldProps = {
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:69]: Move this before the `renderRepeatableActionsField` function, please, for consistency

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  hasRepeatableFields: true,
+  repeatableFieldAction: REPEATABLE_ACTIONS.EXTEND_EXISTING,
+  onRepeatableActionChange: noop,
+};
+
+describe('RepeatableActionsField component', () => {
+  it('should be rendered with child component', () => {
+    const { getByText } = renderRepeatableActionsField(
+      { ...repeatableActionsFieldProps },
+    );
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:87]:
```suggestion
    const { getByText } = renderRepeatableActionsField(repeatableActionsFieldProps);
```

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+    expect(getByText('Repeatable field component'))
+      .toBeDefined();
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:90]:
```suggestion
    expect(getByText('Repeatable field component')).toBeDefined();
```

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+  describe('when click on Select action', () => {
+    describe('if selected action and clear action are equal', () => {
+      it('Add button should be disabled', () => {
+        const {
+          container,
+          getByText,
+        } = renderRepeatableActionsField({ ...repeatableActionsFieldProps });
+        const select = container.querySelector('select');
+        const button = getByText('+Add');
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:101]:
```suggestion
        const addFieldButton = getByText('+Add');
```

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  });
+
+  describe('when click on Select action', () => {
+    describe('if selected action and clear action are equal', () => {
+      it('Add button should be disabled', () => {
+        const {
+          container,
+          getByText,
+        } = renderRepeatableActionsField({ ...repeatableActionsFieldProps });
+        const select = container.querySelector('select');
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:100]:
```suggestion
        const selectRepeatableAction = container.querySelector('select');
```

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        const button = getByText('+Add');
+
+        fireEvent.change(select, { target: { value: 'DELETE_EXISTING' } });
+        expect(button)
+          .toBeDisabled();
```
Comment (mariia-aloshyna) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:105]:
```suggestion
        expect(button).toBeDisabled();
```

File: src/components/RepeatableActionsField/RepeatableActionsField.test.js [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+import '../../../test/jest/__mock__';
+
+import {
+  MAPPING_REPEATABLE_FIELD_ACTIONS, REPEATABLE_ACTIONS,
+} from '../../utils';
+
+import {
+  renderWithReduxForm, translationsProperties,
+} from '../../../test/jest/helpers';
```
Comment (OleksandrHladchenko1) [inline src/components/RepeatableActionsField/RepeatableActionsField.test.js:17]:
```suggestion
import {
  renderWithReduxForm,
  translationsProperties,
} from '../../../test/jest/helpers';
```

**Review by OleksandrHladchenko1 — COMMENTED** [review]
Add description to CHANGELOG.md, please

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Pull request merged**